File: VB/WindowsApplication1/SerializableFormatCondition.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports System.Collections
Imports DevExpress.XtraTreeList.StyleFormatConditions
Imports DevExpress.Utils
Imports DevExpress.XtraGrid
Imports DevExpress.Utils.Serializing

Namespace WindowsApplication1
	Public Class SerializableFormatCondition

		Private _appearance As New AppearanceObject()

		Private _applyToRow As Boolean

		Private _columnName As String = ""

		Private _condition As FormatConditionEnum

		Private _value1 As Object

		Private _value2 As Object

		Private _expression As Object


		<XtraSerializableProperty(XtraSerializationVisibility.Content)> _
		Public Property Appearance() As AppearanceObject
			Get
				Return _appearance
			End Get
			Set(ByVal value As AppearanceObject)
				_appearance.Assign(value)
			End Set
		End Property

		<XtraSerializableProperty()> _
		Public Property ColumnName() As String
			Get
				Return _columnName
			End Get
			Set(ByVal value As String)
				_columnName = value
			End Set
		End Property

		<XtraSerializableProperty()> _
		Public Property Condition() As FormatConditionEnum
			Get
				Return _condition
			End Get
			Set(ByVal value As FormatConditionEnum)
				_condition = value
			End Set
		End Property

		<XtraSerializableProperty()> _
		Public Property ApplyToRow() As Boolean
			Get
				Return _applyToRow
			End Get
			Set(ByVal value As Boolean)
				_applyToRow = value
			End Set
		End Property

		<XtraSerializableProperty()> _
		Public Property Value1() As Object
			Get
				Return _value1
			End Get
			Set(ByVal value As Object)
				_value1 = value
			End Set
		End Property

		<XtraSerializableProperty()> _
		Public Property Value2() As Object
			Get
				Return _value2
			End Get
			Set(ByVal value As Object)
				_value2 = value
			End Set
		End Property
		<XtraSerializableProperty()> _
		Public Property Expression() As Object
			Get
				Return _expression
			End Get
			Set(ByVal value As Object)
				_expression = value
			End Set
		End Property
		Public Sub CopyFromStyleFormat(ByVal style As StyleFormatCondition)
			Appearance = style.Appearance
			ApplyToRow = style.ApplyToRow
			If style.Column Is Nothing Then
				ColumnName = String.Empty
			Else
				ColumnName = style.Column.Name
			End If
			Condition = style.Condition
			Value1 = style.Value1
			Value2 = style.Value2
			Expression = style.Expression
		End Sub

		Public Sub CopyToStyleFormat(ByVal style As StyleFormatCondition)
			style.Appearance.Assign(Appearance)
			style.ApplyToRow = ApplyToRow
			style.Column = (CType(style.Collection, StyleFormatConditionCollection)).TreeListControl.Columns.ColumnByName(ColumnName)
			style.Condition = Condition
			style.Value1 = Value1
			style.Value2 = Value2
			style.Expression = Expression.ToString()
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/SerializableFormatCondition.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using System.Collections;
using DevExpress.XtraTreeList.StyleFormatConditions;
using DevExpress.Utils;
using DevExpress.XtraGrid;
using DevExpress.Utils.Serializing;

namespace WindowsApplication1
{
    public class SerializableFormatCondition
    {

        private AppearanceObject _appearance = new AppearanceObject();

        private bool _applyToRow;

        private string _columnName = "";

        private FormatConditionEnum _condition;

        private object _value1;

        private object _value2;

        private object _expression;


        [XtraSerializableProperty(XtraSerializationVisibility.Content)]
        public AppearanceObject Appearance
        {
            get { return _appearance; }
            set { _appearance.Assign(value); }
        }

        [XtraSerializableProperty()]
        public string ColumnName
        {
            get { return _columnName; }
            set { _columnName = value; }
        }

        [XtraSerializableProperty()]
        public FormatConditionEnum Condition
        {
            get { return _condition; }
            set { _condition = value; }
        }

        [XtraSerializableProperty()]
        public bool ApplyToRow
        {
            get { return _applyToRow; }
            set { _applyToRow = value; }
        }

        [XtraSerializableProperty()]
        public object Value1
        {
            get { return _value1; }
            set { _value1 = value; }
        }

        [XtraSerializableProperty()]
        public object Value2
        {
            get { return _value2; }
            set { _value2 = value; }
        }
        [XtraSerializableProperty()]
        public object Expression
        {
            get { return _expression; }
            set { _expression = value; }
        }
        public void CopyFromStyleFormat(StyleFormatCondition style)
        {
            Appearance = style.Appearance;
            ApplyToRow = style.ApplyToRow;
            ColumnName = style.Column == null? string.Empty:  style.Column.Name;
            Condition = style.Condition;
            Value1 = style.Value1;
            Value2 = style.Value2;
            Expression = style.Expression;
        }

        public void CopyToStyleFormat(StyleFormatCondition style)
        {
            style.Appearance.Assign(Appearance);
            style.ApplyToRow = ApplyToRow;
            style.Column = ((StyleFormatConditionCollection)style.Collection).TreeListControl.Columns.ColumnByName(ColumnName);
            style.Condition = Condition;
            style.Value1 = Value1;
            style.Value2 = Value2;
            style.Expression = Expression.ToString();
        }
    }
}